lib/application.js

```diff
@@ -225,9 +224,10 @@ function respond(ctx) {
     return res.end();
   }
 
-  if ('HEAD' == ctx.method) {
-    if (!res.headersSent && isJSON(body)) {
-      ctx.length = Buffer.byteLength(JSON.stringify(body));
+  if ('HEAD' === ctx.method) {
+    if (!res.headersSent) {
```

Comment:
please add a test case with plain text

Reply:
actually we already set `ctx.response.length` for plain text in [set body()](https://github.com/koajs/koa/blob/5e22f9c8a02d23bf69d76a6f2c03ca69e1b11841/lib/response.js#L135), only expect for object body(for performance reason).